integration-tests/src/test/java/uk/gov/di/authentication/api/IPVCallbackHandlerIntegrationTest.java

```diff
@@ -397,6 +398,66 @@ void shouldStoreReturnCodesInDynamoWhenTheyArePresent(boolean validLoC)
         assertThat(returnCode.size(), equalTo(1));
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void shouldBypassSPoTAndReturnAuthCodeIfIPVReturnsP0ButReturnCodeIsPresentAndRequested(
```

Comment:
I think this test name is slightly misleading as it covers a case when `returnCodeRequested` is `false`. I'd split into two tests and extract some common logic if needed.